botocore/credentials.py

```diff
@@ -63,6 +63,18 @@ def create_credential_resolver(session):
                 num_attempts=num_attempts)
         )
     ]
+
+    # We use ``session.profile`` for EnvProvider rather than
+    # ``profile_name`` because it is ``None`` when unset.
+    if session.profile is None:
+        # No profile has been explicitly set, so we prepend the environment
+        # variable provider. That provider, in turn, may set a profile
+        # or credentials.
+        providers.insert(0, EnvProvider())
+    else:
+        logger.info('Skipping environment variable credential check'
```

Comment:
This seems more like a debug message.  We'll still log which provider ultimately gives creds, but the fact that we're disabling a check seems mostly of interest to a dev.